optuna/integration/botorch.py

```diff
@@ -592,7 +658,9 @@ def _get_default_candidates_func(
     ],
     "torch.Tensor",
 ]:
-    if n_objectives > 3:
+    if n_objectives > 3 and not has_constraint and not consider_running_trials:
+        return ehvi_candidates_func
+    elif n_objectives > 3:
```

Comment:
@contramundum53 is this the kind of condition we want to trigger using EHVI? Let me know, and I'll update tests accordingly.

Reply:
Yes, this is correct. Thanks!